pysat/tests/test_registry.py

```diff
@@ -36,58 +40,385 @@ def create_fake_module(full_module_name, platform, name):
+        """Test error raised when removing module not present"""
+
+        # register all modules at once
+        registry.register(self.module_names)
+
+        # remove non-registered modules using platform and name
+        with pytest.raises(ValueError):
+            registry.remove(platforms, names)
+
+        return
+
+    @pytest.mark.parametrize("platforms, names",
+                             [(['made_up_name', 'second'], ['made_up_name']),
+                              (['made_up_name'], ['made_up_name', 'second']),
+                              ([], ['made_up_name', 'second']),
+                              ([], ['made_up_name'])])
+    def test_platform_name_length_removal_error(self, platforms, names):
+        """Test error raised when platforms and names unequal lengths"""
+
+        # register all modules at once
+        registry.register(self.module_names)
+
+        # unequal lengths
+        with pytest.raises(ValueError):
+            registry.remove(platforms, names)
+
+        return
+
+    def test_module_registration_single(self):
+        """Test registering a module containing an instrument"""
+
+        # register package by module
+        for module in self.modules:
+            root = module[0].split('.')[0]
+            registry.register_by_module(sys.modules[root])
+        # verify instantiation
+        verify_platform_name_instantiation(self.modules)
+        # check that global registry was updated
+        ensure_live_registry_updated(self.modules)
+        # verify update
+        ensure_updated_stored_modules(self.modules)
+
+        return
+
+    def test_module_registration_non_importable(self):
+        """Test registering a non-existent module"""
+
+        with pytest.raises(Exception):
+            registry.register(['made.up.module'])
+
+        return
+
+
+class TestModuleRegistration():
+    def setup(self):
+
+        self.inst_module = instruments
+        # package name
+        pkg_name = self.inst_module.__name__
+
+        # construct inputs similar to TestRegistration
+        # to enable use of existing methods
+        # almost fully general
+        module_names = self.inst_module.__all__
+        self.platform_names = [snip.split('pysat')[-1][1:] for snip in
+                               module_names]
+        self.platforms = ['pysat'] * len(self.platform_names)
+        module_strings = ['.'.join((pkg_name, name)) for name in
+                          module_names]
+        self.modules = [(mod, plat, nam) for mod, plat, nam in
+                        zip(module_strings, self.platforms,
+                            self.platform_names)]
+
+        # remove any existing support which may be let over
+        # errors if you try to remove a package that isn't
+        # registered
+        for plat_name in np.unique(self.platforms):
+            try:
+                registry.remove([plat_name], [None])
+            except ValueError:
+                pass
```

Comment:
Testing may benefit from raising a warning for platforms that can't be removed.

Reply:
The ValueError was raised if someone tried to remove a platform that wasn't registered. I removed that error.

 logger.info now gets a message that the package wasn't registered.